--pretty flag for build mode (for when tsc in not running in a tty)

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Please read the FAQ first, especially the "Common Feature Requests" section.

-->

## Search Terms

<!-- List of keywords you searched for before creating this issue. Write them down here so that others can find this suggestion more easily -->

- pretty tty
- istty
- build pretty

## Suggestion

<!-- A summary of what you'd like to see added or changed -->

I'd like to force pretty outputs in build-mode, even if tsc is not running in a tty.

Currently only `process.stdout.isTTY` determines wether the output in build-mode will use the pretty option or not, and there is no way to overwrite this behaviour (at least none that i know of), as the `--pretty` flag only works in non-build-mode

## Use Cases

<!--
What do you want to use this for?
What shortcomings exist with current approaches?
-->

I have a monorepo-setup and would like to use a combination of "build-mode" "watch-mode" and "restart my application on successful build".

using [tsc-watch](https://github.com/gilamran/tsc-watch) (which basically spawns tsc and listens to the output) I do get exactly that, but with the caveat that the logs from tsc are now no longer in pretty mode, because tsc is not running in a tty.

Having the log-output in pretty mode would be benifical in this scenario, as it would be easier for a developer to spot errors. Having a `--pretty` flag for build-mode would allow just that, because tsc-watch just passes its arguments to tsc.

## Examples

<!-- Show how this would be used and what the behavior would be -->

**build + watch + restart app without pretty-flag:**
`tsc-watch --build --onSuccess 'npm start --prefix server'`
![2019-03-08__23-39-39](https://user-images.githubusercontent.com/20770029/54060027-7fe7b880-41fb-11e9-89c7-5f40076da66f.png)

**build + watch + restart app with pretty-flag:**
`tsc-watch --build --pretty --onSuccess 'npm start --prefix server'`
![2019-03-08__23-38-31](https://user-images.githubusercontent.com/20770029/54059984-5595fb00-41fb-11e9-99c3-116b7f6834e3.png)

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

This may be issue with the project you are using to build your changes. With tsc `--pretty` is allowed and it will show you pretty result.
```
[10:24:04] Starting compilation in watch mode...

../../src/compiler/builder.ts:3:49 - error TS2304: Cannot find name 'ReusableDiagnosticRelatedInformation'.

3     export interface ReusableDiagnostic extends ReusableDiagnosticRelatedInformation {
                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../../src/compiler/builder.ts:7:30 - error TS2304: Cannot find name 'ReusableDiagnosticRelatedInformation'.

7         relatedInformation?: ReusableDiagnosticRelatedInformation[];
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../../src/compiler/builder.ts:271:64 - error TS2304: Cannot find name 'ReusableDiagnosticRelatedInformation'.

271     function convertToDiagnosticRelatedInformation(diagnostic: ReusableDiagnosticRelatedInformation, newProgram: Program): DiagnosticRelatedInformation {
                                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../../src/compiler/builder.ts:654:103 - error TS2304: Cannot find name 'ReusableDiagnosticRelatedInformation'.

654     function convertToReusableDiagnosticRelatedInformation(diagnostic: DiagnosticRelatedInformation): ReusableDiagnosticRelatedInformation {
                                                                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

[10:24:13] Found 4 errors. Watching for file changes.

```

--- a/src/tsc/tsc.ts
+++ b/src/tsc/tsc.ts
@@ -13,7 +13,7 @@ namespace ts {
     }
 
     let reportDiagnostic = createDiagnosticReporter(sys);
-    function updateReportDiagnostic(options?: CompilerOptions) {
+    function updateReportDiagnostic(options: CompilerOptions | BuildOptions) {
         if (shouldBePretty(options)) {
             reportDiagnostic = createDiagnosticReporter(sys, /*pretty*/ true);
         }
@@ -23,7 +23,7 @@ namespace ts {
         return !!sys.writeOutputIsTTY && sys.writeOutputIsTTY();
     }
 
-    function shouldBePretty(options?: CompilerOptions) {
+    function shouldBePretty(options: CompilerOptions | BuildOptions) {
         if (!options || typeof options.pretty === "undefined") {
             return defaultIsPretty();
         }
@@ -192,7 +192,7 @@ namespace ts {
         }
 
         // Update to pretty if host supports it
-        updateReportDiagnostic();
+        updateReportDiagnostic(buildOptions);
         if (projects.length === 0) {
             printVersion();
             printHelp(buildOpts, "--build ");
@@ -209,8 +209,8 @@ namespace ts {
 
         // Use default createProgram
         const buildHost = buildOptions.watch ?
-            createSolutionBuilderWithWatchHost(sys, /*createProgram*/ undefined, reportDiagnostic, createBuilderStatusReporter(sys, shouldBePretty()), createWatchStatusReporter()) :
-            createSolutionBuilderHost(sys, /*createProgram*/ undefined, reportDiagnostic, createBuilderStatusReporter(sys, shouldBePretty()), createReportErrorSummary(buildOptions));
+            createSolutionBuilderWithWatchHost(sys, /*createProgram*/ undefined, reportDiagnostic, createBuilderStatusReporter(sys, shouldBePretty(buildOptions)), createWatchStatusReporter(buildOptions)) :
+            createSolutionBuilderHost(sys, /*createProgram*/ undefined, reportDiagnostic, createBuilderStatusReporter(sys, shouldBePretty(buildOptions)), createReportErrorSummary(buildOptions));
         updateCreateProgram(buildHost);
         buildHost.afterProgramEmitAndDiagnostics = (program: BuilderProgram) => reportStatistics(program.getProgram());
 
@@ -316,7 +316,7 @@ namespace ts {
         };
     }
 
-    function createWatchStatusReporter(options?: CompilerOptions) {
+    function createWatchStatusReporter(options: CompilerOptions | BuildOptions) {
         return ts.createWatchStatusReporter(sys, shouldBePretty(options));
     }
 

